stockflux-securities-master/src/components/SecuritiesTable.js

```diff
@@ -2,39 +2,66 @@ import React, { useEffect, useState } from "react";
 import Components from "stockflux-components";
 import "./SecuritiesTable.css";
 import { Link } from "react-router-dom";
-import { getSecuritiesData } from "../services/SecuritiesService";
+import {
+  getSecuritiesData,
+  deleteSecurity
+} from "../services/SecuritiesService";
+import ValidationError from "../services/ValidationError";
 import AddSecurityButton from "./AddSecurityButton";
 import Alert from "./Alert";
+import TableBody from "./TableBody";
+import { tableEnum } from "../enums";
 
-const SecuritiesTable = () => {
+const SecuritiesTable = ({ match }) => {
   const [securitiesData, setSecuritiesData] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
-  const [errorMessages, setErrorMessages] = useState([]);
-  const timeoutMessage =
-    "Error, unable to get securities data. Please try again!";
+  const [messages, setMessages] = useState([]);
+  const [tableState, setTableState] = useState(tableEnum.loading);
 
-  useEffect(() => {
-    setIsLoading(true);
+  const errorHandler = err => {
+    setTableState(tableEnum.error);
+    if (err instanceof ValidationError) {
+      setMessages(err.messages);
+    } else {
+      setMessages([err.message]);
+    }
+  };
+
+  const getSecuritiesHandler = messages => {
     getSecuritiesData()
       .then(securities => {
-        setIsLoading(false);
         setSecuritiesData(securities);
-        setErrorMessages([]);
+        setTableState(tableEnum.success);
+        setMessages(messages);
       })
-      .catch(() => {
-        setIsLoading(false);
-        setErrorMessages([timeoutMessage]);
+      .catch(err => {
+        setSecuritiesData([]);
+        errorHandler(err);
       });
+  };
+
+  useEffect(() => {
+    setTableState(tableEnum.loading);
+    getSecuritiesHandler(!!match.params.message ? [match.params.message] : []);
   }, []);
 
+  const onClickDelete = securityId => {
+    setTableState(tableEnum.deleting);
+    setMessages([]);
+    deleteSecurity(securityId)
+      .then(response => {
+        getSecuritiesHandler([response.message]);
+      })
+      .catch(err => {
+        errorHandler(err);
+      });
+  };
+
   return (
     <div className="securities-container">
       <div className="securities-title-container">
         <div className="securities-title">Securities</div>
```

Comment:
Shouldn't this be a heading or a paragraph?

Reply:
You mean instead of a div?

Reply:
Is it good practice to wrap text in <h> or <p> tags normally then?

Reply:
I've made the headings for the table and the form into `<h1>` tags and modified the styling

Reply:
> Is it good practice to wrap text in `<h>` or `<p>` tags normally then?

yes